compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEFuseConsumerIntoLoop.cpp

```diff
@@ -0,0 +1,743 @@
+  AMDAIEFuseConsumerIntoLoopPass(const AMDAIEFuseConsumerIntoLoopPass &pass) {}
+  AMDAIEFuseConsumerIntoLoopPass(
+      const AMDAIEFuseConsumerIntoLoopOptions &options)
+      : AMDAIEFuseConsumerIntoLoopBase(options) {}
+
+  void getDependentDialects(DialectRegistry &registry) const override {
+    registry.insert<scf::SCFDialect>();
+  }
+  void runOnOperation() override;
+};
+
+void AMDAIEFuseConsumerIntoLoopPass::runOnOperation() {
+  MLIRContext *context = &getContext();
+  mlir::FunctionOpInterface funcOp = getOperation();
+  IRRewriter rewriter(context);
+
+  // Walk through the graph in post order and find the loop.
+  Operation *scfLoopOp = nullptr;
+  funcOp->walk<WalkOrder::PostOrder, ReverseIterator>(
+      [&](LoopLikeOpInterface op) {
+        if (isa<scf::ForOp>(op) && useSCFFor) {
+          scfLoopOp = op;
+          return WalkResult::interrupt();
+        } else if (isa<scf::ForallOp>(op) && !useSCFFor) {
+          scfLoopOp = op;
+          return WalkResult::interrupt();
+        }
+        return WalkResult::advance();
+      });
+
+  if (!scfLoopOp) {
+    LLVM_DEBUG(llvm::dbgs()
+               << "There is no scf.for/forall loop to fuse with.\n");
+    return;
+  }
+
+  LoopLikeOpInterface loops = cast<LoopLikeOpInterface>(scfLoopOp);
+  auto loopBody = useSCFFor ? cast<scf::ForOp>(loops).getBody()
+                            : cast<scf::ForallOp>(loops).getBody();
+
+  // Search the compute op and its consumer slices.
+  linalg::LinalgOp linalgOp;
+  loopBody->walk<WalkOrder::PostOrder, ReverseIterator>(
+      [&](linalg::LinalgOp op) {
+        if (linalg::isaContractionOpInterface(op)) {
+          linalgOp = op;
+          return WalkResult::interrupt();
+        }
+        return WalkResult::advance();
+      });
+
+  if (!linalgOp) {
+    LLVM_DEBUG(llvm::dbgs()
+               << "----- Could not find any linalg op implementing "
+                  "ContractionOpInterface \n");
+    return;
+  }
+
+  Value::user_range users = linalgOp->getResult(0).getUsers();
+  if (!llvm::hasSingleElement(users)) {
+    LLVM_DEBUG(llvm::dbgs()
+               << "----- Expected only one user of the compute op : "
+               << linalgOp << "\n");
+    return;
+  }
+
+  Operation *terminatorStoreOp = *(users.begin());
+  if (!(isa<tensor::InsertSliceOp, tensor::ParallelInsertSliceOp>(
+          terminatorStoreOp))) {
+    LLVM_DEBUG(llvm::dbgs()
+               << "----- Expected either tensor.insert_slice OR "
+                  "tensor.parallel_insert_slice to be the only user of the "
+                  "compute op : "
+               << linalgOp << "\n");
+    return;
+  }
+
+  if (!terminatorStoreOp) {
+    LLVM_DEBUG(llvm::dbgs()
+               << "----- Consumer ops are already fused it seems \n");
+    return;
+  }
+
+  std::optional<SCFFuseConsumerOfSliceResult> fusedConsumer =
+      tileAndFuseConsumerOfSlice(rewriter, terminatorStoreOp);
+  if (!fusedConsumer) {
+    scfLoopOp->emitWarning(
```

Comment:
wait we dont want the compiler emiting warning everytime it sees a gemm with no elemtwise in the end, so we should just have a debug print here. I am in favor of the test you added to show things work for this case though.